core/security_group_rule.go

```diff
@@ -0,0 +1,191 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type SecurityGroupRule struct {
+	ID        string   `json:"id,omitempty"`
+	Direction string   `json:"direction,omitempty"`
+	Protocol  string   `json:"protocol,omitempty"`
+	Ports     string   `json:"ports,omitempty"`
+	Targets   []string `json:"targets,omitempty"`
+	Notes     string   `json:"notes,omitempty"`
+}
+
+func (sgr *SecurityGroupRule) Ref() SecurityGroupRuleRef {
+	return SecurityGroupRuleRef{ID: sgr.ID}
+}
+
+type SecurityGroupRuleRef struct {
+	ID string `json:"id,omitempty"`
+}
+
+func (sgr SecurityGroupRuleRef) queryValues() *url.Values {
+	v := &url.Values{}
+	v.Set("security_group_rule[id]", sgr.ID)
+
+	return v
+}
+
+type SecurityGroupRuleArguments struct {
+	Direction string   `json:"direction,omitempty"`
+	Protocol  string   `json:"protocol,omitempty"`
+	Ports     string   `json:"ports,omitempty"`
+	Targets   []string `json:"targets,omitempty"`
+	Notes     string   `json:"notes,omitempty"`
```

Comment:
Is it possible to specifically set `Targets` or `notes` to a empty list (remove existing targets), and empty string (clear out existing notes value)? If so these will need to be pointer types so we can differentiate between unchanged and clear updates.